export/securedrop_export/print/service.py

```diff
@@ -220,43 +243,131 @@ def _print_all_files(self):
             self._print_file(file_path)
             logger.info(f"Printing document {print_count} of {len(files)}")
 
-    def _is_open_office_file(self, filename):
-        OPEN_OFFICE_FORMATS = [
-            ".doc",
-            ".docx",
-            ".xls",
-            ".xlsx",
-            ".ppt",
-            ".pptx",
-            ".odt",
-            ".ods",
-            ".odp",
-            ".rtf",
+    def _get_supported_mimetypes_libreoffice(self):
+        """
+        Return a list of mimetypes supported by Libreoffice programs.
+        """
+        supported_mimetypes = []
+        prefix = "/usr/share/applications/"
+        libreoffice_programs = [
+            "libreoffice-base",
+            "libreoffice-calc",
+            "libreoffice-draw",
+            "libreoffice-impress",
+            "libreoffice-math",
+            "libreoffice-writer",
         ]
-        for extension in OPEN_OFFICE_FORMATS:
-            if os.path.basename(filename).endswith(extension):
-                return True
-        return False
+        for item in libreoffice_programs:
+            desktop_file = Path(prefix, item, ".desktop")
+            if desktop_file.exists():
+                with open(desktop_file) as f:
+                    line = f.readline()
+                    if line.startswith("MimeType="):
+                        # Semicolon-separated list; don't leave empty element at the end
+                        supported_mimetypes.append(line.strip("MimeType=").split(";")[:-1])
+
+        return supported_mimetypes
+
+    def _needs_pdf_conversion(self, filename):
+        """
+        Checks mimetype of a file and returns True if file must be converted
+        to PDF before attempting to print.
+
+        Raises ExportException for unprintable mimetypes or on mimetype
+        discovery error.
+        """
+        mimetype = None
+        supported_types = []
+
+        try:
+            supported_types = self._get_supported_mimetypes_libreoffice()
+        except OSError as e:
+            logger.error(f"Could not get supported mimetypes list: {e}")
+            raise ExportException(sdstatus=Status.ERROR_MIMETYPE_DISCOVERY)
+        if len(supported_types) == 0:
+            raise ExportException(sdstatus=Status.ERROR_MIMETYPE_DISCOVERY)
+
+        try:
+            # b'filename: application/bla\n'
+            mimetype = (
+                subprocess.check_output(["mimetype", filename]).decode().split()[0].strip(":")
```

Comment:
Any difference in using `mimetype` vs `file --mime` or https://pypi.org/project/python-magic/ ?

Reply:
I ended up going with `mimetype -M` (the magic bytes flag). 
Mimetype looks first in XDG_DATA_DIRS (and other places), and `file` looks in /etc/mime I think.